### ISSUE TITLE
Spherical Camera support

### ISSUE THREAD

**Issue by bhouston** [gkjohnson/three-gpu-pathtracer#166]
**Is your feature request related to a problem? Please describe.**

The renderer currently supports both the Perspective and Orthographic camera.  Many production renderers, although not games engines, support so called 360 or Spherical cameras.  These can be incredibly useful for rendering panoramic views.

It would be nice to add this to the renderer.

This would be nice for rendering background scenes into envmaps or creating panoramic viewers of complex scenes that otherwise could not be realized in interactive 3D.

**Describe the solution you'd like**

I am unsure how you would specify a spherical camera in Three.js.  Maybe specify it as a CubeCamera?

The output from this renderer should be a latlong arranged image.  So it should varying from -180 degrees to +180 degrees longitude on the x axis.  And from -90 degrees to +90 degrees latitude on the y axis.

Basically the output should look like this:

![kitchenb_3](https://user-images.githubusercontent.com/588541/171656001-14fded09-5b0d-4648-8eda-84f2a6e0603c.jpg)

**Comment by gkjohnson** [comment]
I think this is probably simplest as a #define like `FEATURE_EQUIRECT_CAMERA` or uniform like the Ortho camera. Then it will just render as an equirect map ignoring the projection matrix. There's already a shader function for going from uv to equirect direction here:

https://github.com/gkjohnson/three-gpu-pathtracer/blob/33936d3eaad18064ba849b6bf4f85e6bc5256682/src/shader/shaderUtils.js#L213

**Comment by JustASquid** [comment]
Looking into this, I have a question for @gkjohnson.

The jittering code looks like this at the moment:

```
vec2 ndc = 2.0 * vUv - vec2( 1.0 );

vec3 ss00 = ndcToRayOrigin( vec2( - 1.0, - 1.0 ) );
vec3 ss01 = ndcToRayOrigin( vec2( - 1.0, 1.0 ) );
vec3 ss10 = ndcToRayOrigin( vec2( 1.0, - 1.0 ) );

vec3 ssdX = ( ss10 - ss00 ) / resolution.x;
vec3 ssdY = ( ss01 - ss00 ) / resolution.y;

// Jitter the camera ray by finding a new subpixel point to point to from the camera origin
// This is better than just jittering the camera position since it actually results in divergent
// rays providing better coverage for the pixel
rayOrigin = ndcToRayOrigin( ndc ) + tentFilter( rand() ) * ssdX + tentFilter( rand() ) * ssdY;

```

In the absence of a projection matrix, this doesn't really work. My natural inclination would be to jitter the input uv coordinates instead. I'd like to keep the implementation clean though, so it seems to make sense to apply the input uv jittering to the other projections if possible. My question is, is there a particular reason why you are jittering the ray origin post-projection instead of the input UV pre-projection?

**Comment by gkjohnson** [comment]
Right - it should be fine to jitter the the uv coordinates by `+- 0.5 / resolution` in each direction which should be simpler to maintain, as well. Thanks for taking a look!

**Comment by bhouston** [comment]
For completeness sake, Threekit did put a bounty on a merged PR of $500.
